### PR TITLE
document COPY TO query parameter

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

**Comment by seut** [comment]
ack

File: docs/sql/reference/copy_to.txt [hunk 1]
```diff
@@ -60,6 +61,11 @@ Parameters
 
 :column: (optional) A list of column expressions that should be exported.
 
+:query: A SELECT statement whose result is copied. Parentheses are required
+        around the query.
+        Note that COPY TO operates on shard level. This means that any queries
+        which require re-distribution are not supported.
```
Comment (dobe) [inline docs/sql/reference/copy_to.txt]: should be documented

File: docs/sql/dml.txt [hunk 2]
```diff
@@ -427,6 +427,14 @@ the table and shard id with gzip compression::
...
     COPY OK, 3 rows affected ...
 
+
+Instead of specifying a table to be exported it is also possible to use a
+SELECT statement to declare the result to be exported. This is useful if only a
+subset of the data needs to be exported::
+
+    cr> copy (select quote from quotes where match(quote, 'time')) TO DIRECTORY '/tmp/';
+    COPY OK, 1 rows affected ...
+
```
Comment (dobe) [inline docs/sql/dml.txt]: the output format is object, probably this should be mentioned?

File: docs/sql/reference/copy_to.txt [hunk 3]
```diff
@@ -12,39 +12,54 @@ Synopsis
...
                      [ WITH ( copy_parameter [= value] [, ... ] ) ]
 
 Description
 ===========
 
-Copy the contents of a table to one or many files on any cluster node
-containing data from the given table.
+The ``COPY TO`` command exports the contents of a table to one or more files.
+Each node of the cluster which contains a shard of the table will export the
+contents of the shards it contains.
```
Comment (mfelsche) [inline docs/sql/reference/copy_to.txt]: maybe ``... of its shards`` to avoid double "contains"

File: docs/sql/reference/copy_to.txt [hunk 4]
```diff
@@ -12,39 +12,54 @@ Synopsis
...
+cause corruption in the file that is being written. The same also holds true
+when data is being written to a shared file system such as `Amazon S3`_ or
+`NFS`_.
+
+In order to prevent this from happening it is required to write to a separate
+file per shard. To do this the ``DIRECTORY`` keyword can be used. An example is
+given in :ref:`exporting_data`.
+
+
+The ``query`` parameter is being limited to queries which can be materialized
```
Comment (dobe) [inline docs/sql/reference/copy_to.txt]: is limited

**Comment by dobe** [comment]
ack

File: docs/sql/dml.txt (hunk 2 above)
Comment (mfussenegger) [inline docs/sql/dml.txt]: It's documented in the reference section

**Comment by adrpar** [comment]
This PR can be closed, since PR #3022 supersedes it.

**Pull request closed without merging**